version: '3.8'

services:
  auth-service:
    image: authservice
    container_name: auth-service
    build:
      context: ./back-end/auth-service/
    restart: on-failure
    networks:
      - app_network
    depends_on:
      rabbitmq:
        condition: service_healthy

  profile-service:
    image: profileservice
    container_name: profile-service
    build:
      context: ./back-end/profile-service/
    restart: on-failure
    networks:
      - app_network
    depends_on:
      rabbitmq:
        condition: service_healthy

  match-service:
    image: matchservice
    container_name: match-service
    build:
      context: ./back-end/match-service/
    restart: on-failure
    networks:
      - app_network
    depends_on:
      rabbitmq:
        condition: service_healthy
       
  apiGateway:
    image: apigateway
    container_name: api-gateway
    build:
      context: ./back-end/apiGateway/
    ports:
      - 7000:80
    networks:
      - app_network

  rabbitmq:
    image: rabbitmq:3-management
    environment:
      RABBITMQ_DELAYED_QUEUE_MODE: "on"
    container_name: rabbitmq
    hostname: rabbitmq
    ports:
      - 15672:15672
      - 5672:5672
    healthcheck:
      test: ["CMD", "rabbitmqctl", "status"]
      interval: 5s
      timeout: 10s
      retries: 2
    restart: on-failure
    networks:
      - app_network
    
networks:
  app_network:
    driver: bridge